src/main/java/net/openhft/chronicle/wire/AbstractGeneratedMethodReader.java

```diff
@@ -243,7 +238,7 @@ public boolean readOne() {
                 context.rollbackOnClose();
```

Comment:
I'm not sure that this part with `rollbackOnClose` makes sense when we don't have a delegate.

Normally, the generated instance should never return `false`, but if it happens, we should skip an event (ideally, with a warning) and go on. Rolling back will force the method reader to spin on such "erroneous" event.

Reply:
Nice one - well spotted - thanks! Totally agree with you re: comment below and extra tests - this needs to be bulletproof

Reply:
Yep - I have now removed the rollbackOnClose. The method reader always returns true if there is some data to be read as there is always a delegate (there is a default one even if you don't specify - see `MethodReaderDelegationTest#testUnsuccessfulCallNoDelegate()`)